optimizely/event/event_processor.py

```diff
@@ -222,3 +222,29 @@ def dispose(self):
       return
 
     self._disposed = True
+
+
+class ForwardingEventProcessor(EventProcessor):
+
+  def __init__(self, event_dispatcher, logger, notification_center=None):
+    self.event_dispatcher = event_dispatcher
+    self.logger = logger
+    self.notification_center = notification_center
+
+  def process(self, event):
+    if not isinstance(event, UserEvent):
+      # TODO: log error
+      return
+
+    log_event = EventFactory.create_log_event(event, self.logger)
+
+    if self.notification_center is not None:
```

Comment:
I think this should be notified after `dispatch_event`.

Reply:
It is how we are doing in Java too. As soon as a log_event is created we send a notification.
@mnoman09 can you please give your input here?